src/Microsoft.SqlTools.ServiceLayer/QueryExecution/Contracts/CopyResultsRequest.cs

```diff
@@ -0,0 +1,56 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+#nullable disable
+
+using Microsoft.SqlTools.Hosting.Protocol.Contracts;
+
+namespace Microsoft.SqlTools.ServiceLayer.QueryExecution.Contracts
+{
+    public class TableSelectionRange
+    {
+        public int FromRow { get; set; }
+        public int ToRow { get; set; }
+        public int FromColumn { get; set; }
+        public int ToColumn { get; set; }
+    }
+
+    /// <summary>
+    /// Parameters for the copy results request
+    /// </summary>
+    public class CopyResultsRequestParams : SubsetParams
+    {
+        /// <summary>
+        /// Whether to remove the line break from cell values.
+        /// </summary>
+        public bool RemoveNewLines { get; set; }
+
+        /// <summary>
+        /// Whether to include the column headers.
+        /// </summary>
+        public bool IncludeHeaders { get; set; }
```

Comment:
is this going to support "Copy only headers" like discussed in this issue https://github.com/microsoft/azuredatastudio/issues/8902?

Reply:
this property is to control whether the column names should be included along with the selected cells. the copy header action is a separate thing and we can already get it from the result metadata directly in ADS.

![image](https://github.com/microsoft/sqltoolsservice/assets/13777222/03495b1e-df6e-467e-bc83-75e916d111db)